fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1044,42 +1049,69 @@ protected SqlQueryData processInclusionCriteria(QueryParameter queryParm) throws
 
-                // split the resource type name out (since issue #1366)
-                String resourceTypeName = null;
-                String[] parts = currentParmValue.split("/");
-                if (parts.length == 2) {
-                    resourceTypeName = parts[0];
-                    currentParmValue = parts[1];
-                }
-
-                // Build this piece: pX.token_value = search-attribute-value [ AND pX.code_system_id = <n> ]
-                whereClauseSegment.append(PARAMETER_TABLE_ALIAS).append(DOT).append(TOKEN_VALUE).append(operator)
-                        .append(BIND_VAR);
-                if (resourceTypeName != null) {
-
+                boolean fallback = false;
+                Set<String> commonTokenValueIds = new HashSet<>();
+                for (String currentParmValue : currentParmValues) {
                     Long commonTokenValueId = getCommonTokenValueId(resourceTypeName, currentParmValue);
-                    if (commonTokenValueId != null) {
-                        // #1929 improves cardinality estimation
-                        // resulting in far better execution plans for many search queries. Because COMMON_TOKEN_VALUE_ID
-                        // is the primary key for the common_token_values table, we don't need the CODE_SYSTEM_ID = ? predicate.
-                        whereClauseSegment.append(AND).append(PARAMETER_TABLE_ALIAS).append(DOT).append(COMMON_TOKEN_VALUE_ID).append(EQ)
-                            .append(commonTokenValueId);
+                    if (commonTokenValueId == null) {
+                        // Can't use the common_token_value_id optimization, so do it the old way.
+                        log.warning("Unable to obtain common token value id for " + resourceTypeName + "/" + currentParmValue);
+                        fallback = true;
+                        break;
                     } else {
-                        // Can't use the common_token_value_id optimization, so do it the old way
-                        // and join against the code-system.
-                        Integer codeSystemIdForResourceType = getCodeSystemId(resourceTypeName);
-                        whereClauseSegment.append(AND).append(PARAMETER_TABLE_ALIAS).append(DOT)
-                        .append(CODE_SYSTEM_ID).append(EQ).append(nullCheck(codeSystemIdForResourceType));
+                        commonTokenValueIds.add(commonTokenValueId.toString());
                     }
                 }
+
+                if (fallback) {
+                    // Can't use the common_token_value_id optimization, so do it the old way.
+                    // pX.token_value IN (val1, val2, ...) [ AND pX.code_system_id = n ]
+                    whereClauseSegment.append(PARAMETER_TABLE_ALIAS).append(DOT).append(TOKEN_VALUE)
+                            .append(IN).append(LEFT_PAREN)
+                            .append(String.join(", ", Collections.nCopies(currentParmValues.size(), BIND_VAR)))
```

Comment:
Good question.  The short answer is that I did it this way because thats what we were already doing:  originally we used parameterization here but when we added support for COMMON_TOKEN_VALUE_ID (#1366), we just used the literal directly (old line 1072).  Was that you or @punktilious?

Anyway, I debated this in my head but decided its fine to keep using the literal strings in the COMMON_TOKEN_VALUE_ID case because those ids are coming from the database (not user-provided).  Hopefully we're always in that case going forward, but I wasn't confident enough to remove the "old way" at this time (because I think use of the new way is still behind a config flag).